driver/src/price_finding/price_finder_interface.rs

```diff
@@ -23,37 +20,36 @@ impl Default for Fee {
 }
 
 #[derive(Clone, Debug, Copy, PartialEq)]
-pub enum SolverType {
+pub enum SolverConfig {
     NaiveSolver,
-    StandardSolver,
-    FallbackSolver,
+    StandardSolver(u32), //params for --solverTimeLimit
+    FallbackSolver(u32),
 }
 
-impl FromStr for SolverType {
-    type Err = Infallible;
-
-    fn from_str(solver_type_str: &str) -> Result<Self, Self::Err> {
+impl SolverConfig {
+    pub fn new(solver_type_str: &str, solver_time_limit: u32) -> Self {
         match solver_type_str.to_lowercase().as_str() {
-            "standard-solver" => Ok(SolverType::StandardSolver),
-            "fallback-solver" => Ok(SolverType::FallbackSolver),
+            "standard-solver" => SolverConfig::StandardSolver(solver_time_limit),
+            "fallback-solver" => SolverConfig::FallbackSolver(solver_time_limit),
             // the naive solver is the standard solver.
-            _ => Ok(SolverType::NaiveSolver),
+            _ => SolverConfig::NaiveSolver,
```

Comment:
I think it's more safe to panic in the default case (since this happens on startup) and require people to actually use valid configs. Otherwise a small type in standard-sovler will give you the naive solver without a warning.